pkg/moov/transfer_api.go

```diff
@@ -31,10 +32,36 @@ func WithTransferIdempotencyKey(key uuid.UUID) CreateTransferArgs {
 	}
 }
 
+type CreateTransferTimeoutError struct {
+	TransferStarted TransferStarted
+}
+
+func (e CreateTransferTimeoutError) Error() string {
+	return fmt.Sprintf("created the transfer with transferID=%v, but timeout occured while waiting for the synchronous response with rail-specific details", e.TransferStarted.TransferID)
+}
+
+func (c Client) CreateTransferSync(ctx context.Context, create CreateTransfer, options ...CreateTransferArgs) (*Transfer, error) {
```

Comment:
I don't like it broken out like this. Its the same endpoint with just a different option for waiting. You're also removing the optionality of where to WaitFor for future features.

Reply:
I also don't agree with getting rid of the third argument for async timeout case. Having that async timeout in the return keeps one from having to dig it out of the error response or somewhere else. 

Its not an error its a timeout for something that will complete, its not an end state. Leaving that hidden inside of the error could lead to a user of the library to just consider it failed and that its not going to complete. The same way error is returned on everything versus say panic is to show people this is something that needs to be resolved and remedied.

Reply:
The 2nd return value representing progress makes sense to me. We can explain it's (model, progress, error) where `error` is returned if the request didn't get converted into model & progress. Either from a client side problem, network error, server rejection, etc. If we're consistent then it's predictable and easier to explain.

Reply:
That sounds easy to explain/understand and great to document.

Reply:
Appreciate the feedback, makes sense.

Was mainly trying to avoid the ambiguous nature of returning 3 params, and not fully knowing whether `Transfer` or `TransferStarted` will be set; which is dependent on the inputs and response code (200/201/202) -- then forcing the user to dig into our docs.

Currently, I don't love either approach, but was wondering if there was a lesser of two evils.

 Thinking about this more, I have one more alternative: https://github.com/moovfinancial/moov-go/pull/96